ultralytics/yolo/utils/plotting.py

```diff
@@ -132,6 +143,45 @@ def masks(self, masks, colors, im_gpu, alpha=0.5, retina_masks=False):
             # convert im back to PIL and update draw
             self.fromarray(self.im)
 
+    def kpts(self, kpts, shape=(640, 640), steps=3, radius=5, kpt_line=True):
+        """Plot keypoints.
+        Args:
+            kpts (tensor): predicted kpts, shape: [51]
+            shape (tuple): image shape, (h, w)
+            steps (int): keypoints step
+            radius (int): size of drawing points
+        """
+        if self.pil:
+            # convert to numpy first
+            self.im = np.asarray(self.im).copy()
+        num_kpts = len(kpts) // steps
+        for kid in range(num_kpts):
+            x_coord, y_coord = kpts[steps * kid], kpts[steps * kid + 1]
+            if x_coord % shape[1] != 0 and y_coord % shape[0] != 0:
+                if steps == 3:
+                    conf = kpts[steps * kid + 2]
+                    if conf < 0.5:
```

Comment:
@Laughing-q conf here is visibility right? The 3rd co-ordinate in n,17,3 ?

Reply:
don't worry, I handled this in the landmark branch.

Reply:
ok got it